src/internal/packager/images/pull.go

```diff
@@ -7,506 +7,393 @@ package images
-
-					if dockerEndPointHost == "" {
-						dockerEndPointHost, err = getDockerEndpointHost()
-						if err != nil {
-							return err
-						}
-					}
-					// Attempt to connect to the local docker daemon.
-					cli, err := client.NewClientWithOpts(
-						client.WithHost(dockerEndPointHost),
-						client.WithTLSClientConfigFromEnv(),
-						client.WithVersionFromEnv(),
-					)
-					if err != nil {
-						return fmt.Errorf("docker not available: %w", err)
-					}
-					cli.NegotiateAPIVersion(ectx)
-
-					// Inspect the image to get the size.
-					rawImg, _, err := cli.ImageInspectWithRaw(ectx, ref)
-					if err != nil {
-						return err
-					}
-
-					// Warn the user if the image is large.
-					if rawImg.Size > 750*1000*1000 {
-						message.Warnf("%s is %s and may take a very long time to load via docker. "+
-							"See https://docs.zarf.dev/faq for suggestions on how to improve large local image loading operations.",
-							ref, utils.ByteFormat(float64(rawImg.Size), 2))
-					}
-
-					// Use unbuffered opener to avoid OOM Kill issues https://github.com/zarf-dev/zarf/issues/1214.
-					// This will also take forever to load large images.
-					img, err = daemon.Image(reference, daemon.WithUnbufferedOpener(), daemon.WithClient(cli))
-					if err != nil {
-						return fmt.Errorf("failed to load from docker daemon: %w", err)
-					}
-				} else {
-					img, err = crane.Pull(ref, opts...)
-					if err != nil {
-						return fmt.Errorf("unable to pull image %s: %w", refInfo.Reference, err)
-					}
-				}
-			}
-
-			if err := checkForIndex(refInfo, desc); err != nil {
-				return err
 			}
 
-			cacheImg, err := utils.OnlyHasImageLayers(img)
+			localRepo.Reference, err = registry.ParseReference(overriddenRef)
 			if err != nil {
 				return err
 			}
-			if cacheImg && cfg.CacheDirectory != "" {
-				img = cache.Image(img, cache.NewFilesystemCache(cfg.CacheDirectory))
-			}
 
-			size, err := getSizeOfImage(img)
-			if err != nil {
-				return fmt.Errorf("failed to get size of image: %w", err)
-			}
-			totalBytes.Add(size)
+			localRepo.Client = client
 
-			layers, err := img.Layers()
-			if err != nil {
-				return fmt.Errorf("unable to get layers for %s: %w", refInfo.Reference, err)
+			// If the image has a digest start out by checking if it's an index sha
+			if image.Digest != "" {
+				err := checkForIndex(ectx, localRepo, overriddenRef, image)
+				if err != nil {
+					return err
+				}
 			}
 
-			shaLock.Lock()
-			defer shaLock.Unlock()
-			for _, layer := range layers {
-				digest, err := layer.Digest()
-				if err != nil {
-					return fmt.Errorf("unable to get digest for image layer: %w", err)
+			fetchOpts := oras.DefaultFetchBytesOptions
+			fetchOpts.FetchOptions.TargetPlatform = platform
+			desc, b, err := oras.FetchBytes(ectx, localRepo, overriddenRef, fetchOpts)
+			if err != nil {
+				if strings.Contains(err.Error(), "toomanyrequests") {
```

Comment:
Add a TODO here, we should consider a backoff logic for those cases. k8s has several primitives [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/wait/backoff.go) that we cold just re-use.